kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestAvroWriter.java

```diff
@@ -16,17 +16,80 @@
 
 package org.kitesdk.data.spi.filesystem;
 
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericData.Record;
+import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.Path;
+import org.junit.Assert;
+import org.junit.Test;
 import org.kitesdk.data.DatasetDescriptor;
+import org.kitesdk.data.DatasetReader;
 import org.kitesdk.data.DatasetWriter;
+import org.kitesdk.data.Flushable;
+import org.kitesdk.data.Syncable;
+import org.kitesdk.data.spi.ReaderWriterState;
 
-public class TestAvroWriter extends TestFileSystemWriters<Object> {
+public class TestAvroWriter extends TestFileSystemWriters {
   @Override
-  public DatasetWriter<Object> newWriter(Path directory) {
-    return new FileSystemWriter<Object>(fs, directory,
+  public FileSystemWriter<Record> newWriter(Path directory, Schema schema) {
+    return FileSystemWriter.newWriter(fs, directory,
         new DatasetDescriptor.Builder()
-            .schemaLiteral("\"string\"")
+            .schema(schema)
             .format("avro")
             .build());
   }
+
+  @Override
+  public DatasetReader<Record> newReader(Path path, Schema schema) {
+    return new FileSystemDatasetReader<Record>(fs, path, schema, Record.class);
+  }
+
+  @Test
+  public void testIsFlushable() {
+    Assert.assertTrue(fsWriter instanceof FileSystemWriter.IncrementalWriter);
```

Comment:
Test for the public interface (org.kitesdk.data.Flushable) rather than the private implementation class.